mandatory/2-function-creation.js

```diff
@@ -25,7 +33,17 @@ Write a function that:
 - removes any forward slashes (/) in each string
 - makes all strings all lowercase
 */
-function tidyUpStrings(arrayOfStrings) {}
+function tidyUpStrings(arrayOfStrings) {
+  let newStrings = []
+  for (let i=0;i<arrayOfStrings.length; i++)
+  {
+    arrayOfStrings[i]=arrayOfStrings[i].trim();
+    arrayOfStrings[i]=arrayOfStrings[i].toLowerCase();
+    arrayOfStrings[i]=arrayOfStrings[i].replace("/","");
+    newStrings.push(arrayOfStrings[i]);
+  }
+  return newStrings;
+}
 
```

Comment:
Interesting! There's something useful to look at here and dig in to. With this method you've set up here you are mutating your original array and then returning a new array. In general, in the [functional programming paradigm](https://syllabus.codeyourfuture.io/guides/paradigm#functional) we teach here, we try to avoid mutation. 

(It's ok to not quite get why this matters yet, but it's useful to know at this point that it does matter.)

Try writing this function again, but this time, instead of mutating the value in arrayOfStrings and then copying that mutated value into newStrings with push, use a temporary variable inside your loop:

`let cleanedString = arrayOfStrings[i]`

and do your operations on that and push into your new array. Take this idea (mutation) to class and talk about it.

Reply:
Done